rero_ils/modules/acq_orders/serializers/csv.py

```diff
@@ -0,0 +1,283 @@
+                    .get_records_by_terms(terms=order_pids,
+                                          key='acq_order__pid')
+                order_lines = {}
+                doc_pids = []
+                account_pids = []
+                for result in records:
+                    order_lines[result.pid] = result.to_dict()
+                    doc_pids.append(result.document.pid)
+                    account_pids.append(result.acq_account.pid)
+
+                docs = get_documents_by_pids(doc_pids)
+                accounts = get_accounts_by_pids(account_pids)
+                receipt_lines = get_receipt_lines_by_order_line_pids(
+                    list(order_lines.keys()))
+                return order_lines, docs, accounts, receipt_lines
+
+            def get_documents_by_pids(doc_pids):
+                """Get documents for the given pids."""
+                def _build_doc(data):
+                    document_data = {
+                        'document_title': next(
+                            filter(lambda x: x.get('type') == 'bf:Title',
+                                   data.get('title'))
+                        ).get('_text')
+                    }
+                    # process provision activity
+                    provision_activity = next(
+                        filter(lambda x: x.get('type') == 'bf:Publication',
+                               data.get('provisionActivity'))
+                    )
+                    start_date = provision_activity.get('startDate', '')
+                    end_date = provision_activity.get('endDate')
+                    document_data['document_publication_year'] = \
+                        f'{start_date} - {end_date}' \
+                        if end_date else start_date
+
+                    document_data['document_publisher'] = ' | '.join(
+                        data['value']
+                        for stmt in provision_activity.get('statement', [])
+                        for data in stmt.get('label', [])
+                        if stmt['type'] == 'bf:Agent'
+                    )
+                    document_data['document_edition_statement'] = ' | '.join(
+                        edition.get('value')
+                        for edition_statement in
+                        data.get('editionStatement', [])
+                        for edition in edition_statement.get('_text', [])
+                    )
+                    document_data['document_series_statement'] = ' | '.join(
+                        data['value']
+                        for serie in data.get('seriesStatement', [])
+                        for data in serie.get('_text', [])
+                    )
+                    document_data['document_isbn'] = ' | '.join(
+                        data.get('isbn', [])
+                    )
+
+                    # process contributions
+                    creator = []
+                    if 'contribution' in data:
```

Comment:
Probably this can be refactored. Try `sorcery`.

Reply:
Done